### PR TITLE
[Refactor] `sample_registers`.

### PR DESCRIPTION
**Pull request by d0cd**
This PR, refactors `sample_registers` into a common `test_helpers` module.

### CODE REVIEW

File: synthesizer/src/program/instruction/operation/assert.rs [hunk 1]
```diff
@@ -241,11 +241,13 @@ impl<N: Network, const VARIANT: u8> ToBytes for AssertInstruction<N, VARIANT> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::{ProvingKey, Registers, Store, StoreCircuit, VerifyingKey};
+
+    use crate::{program::test_helpers::sample_registers, ProvingKey, VerifyingKey};
+
```
Comment (howardwu) [inline synthesizer/src/program/instruction/operation/assert.rs]:
```suggestion
    use crate::{program::test_helpers::sample_registers, ProvingKey, VerifyingKey};
```

File: synthesizer/src/program/instruction/operation/is.rs [hunk 2]
```diff
@@ -253,9 +253,11 @@ impl<N: Network, const VARIANT: u8> ToBytes for IsInstruction<N, VARIANT> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::{ProvingKey, Registers, VerifyingKey};
+
+    use crate::{program::test_helpers::sample_registers, ProvingKey, VerifyingKey};
+
```
Comment (howardwu) [inline synthesizer/src/program/instruction/operation/is.rs]:
```suggestion
    use crate::{program::test_helpers::sample_registers, ProvingKey, VerifyingKey};
```

File: synthesizer/src/program/instruction/operation/mod.rs [hunk 3]
```diff
@@ -769,3 +769,61 @@ crate::operation!(
         (U128, U128) => U128,
     }
 );
+
+#[cfg(test)]
+pub(crate) mod test_helpers {
+    use super::*;
+
+    use crate::{Authorization, CallStack, Registers, Stack, Store, StoreCircuit};
+
```
Comment (howardwu) [inline synthesizer/src/program/instruction/operation/mod.rs]:
```suggestion
    use crate::{Authorization, CallStack, Registers, Stack, Store, StoreCircuit};
```

File: synthesizer/src/program/instruction/operation/assert.rs [hunk 4]
```diff
@@ -263,7 +265,6 @@ mod tests {
         cache: &mut HashMap<String, (ProvingKey<CurrentNetwork>, VerifyingKey<CurrentNetwork>)>,
     ) -> Result<(Stack<CurrentNetwork>, Vec<Operand<CurrentNetwork>>)> {
```
Comment (howardwu) [inline synthesizer/src/program/instruction/operation/assert.rs:265]: Shouldn't `run` be outputted from this method if we're really abstracting it?

Comment (d0cd) [reply synthesizer/src/program/instruction/operation/assert.rs:265]: Agreed, but we're not abstracting `sample_stack`in this PR.
That could be done in a follow up PR, which dedups `sample_stack` across all this tests in `operations`

**Review by howardwu — APPROVED** [review]
LGTM

**Pull request merged**